Content.Server/Chemistry/Metabolism/MovespeedModifierMetabolism.cs

```diff
@@ -0,0 +1,61 @@
+using Content.Shared.Chemistry.Reagent;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Serialization.Manager.Attributes;
+using Content.Server.GameObjects.Components.Chemistry;
+using System.Threading;
+using Content.Shared.Chemistry.Metabolizable;
+using Content.Shared.Movement.Components;
+
+namespace Content.Server.Chemistry.Metabolism
+{
+    /// <summary>
+    /// Default metabolism for stimulants and tranqs. Attempts to find a MovementSpeedModifier on the target,
+    /// adding one if not there and to change the movespeed
+    /// </summary>
+    [DataDefinition]
+    public class MovespeedModifierMetabolism : IMetabolizable
+    {
+
+        /// <summary>
+        /// How much of the reagent should be metabolized each sec.
+        /// </summary> 
+        [DataField("rate")]
+        public ReagentUnit MetabolismRate { get; set; } = ReagentUnit.New(1);
+
+        /// <summary>
+        /// Walk speed modifier
+        /// </summary>
+        [DataField("walkSpeedModifier")]
+        public float WalkSpeedModifier { get; set; } = 10;
+
+        /// <summary>
+        /// Run speed modifier
+        /// </summary>
+        [DataField("sprintSpeedModifier")]
+        public float SprintSpeedModifier { get; set; } = 10;
+
+        private const int StatusLifetime = 1200;
+        /// <summary>
+        /// Remove reagent at set rate, changes damage if a DamageableComponent can be found.
+        /// </summary>
+        /// <param name="solutionEntity"></param>
+        /// <param name="reagentId"></param>
+        /// <param name="tickTime"></param>
+        /// <returns></returns>
+        ReagentUnit IMetabolizable.Metabolize(IEntity solutionEntity, string reagentId, float tickTime)
+        {
+            if (solutionEntity.TryGetComponent(out MovementSpeedModifierComponent? movement))
+            {
+                solutionEntity.EnsureComponent(out MovespeedStatusComponent status);
+
+                status.SprintSpeedModifier = WalkSpeedModifier;
+                status.SprintSpeedModifier = SprintSpeedModifier;
+                status.EffectTime = StatusLifetime;
+                movement.RefreshMovementSpeedModifiers();
```

Comment:
A) You're setting SprintSpeed to WalkSpeed
B) You shouldn't make setting the properties reset the timer because now you're resetting the timer a bunch of times in a row needlessly. You should just manually reset it after.
C) You should multiply the timer by how much is metabolised to make this tickrate-agnostic.